### ISSUE TITLE
divvy plugins idea

### ISSUE THREAD

**Issue by nsheff** [pepkit/divvy#51]
As I've been trying to document looper, divvy and how adapters work, I've struggled to come up with succinct ways to explain things, and had a though...

Right now, divvy ships with looper adapters. This is a bit of an unsatisfying entanglement... I mean, I know it doesn't affect divvy's independent function, but still, it would be nice if divvy was totally independent of looper. But on the other hand, I don't really want to introduce a new step of making people manually add looper adapters into their divvy config file when they install looper.

So, what about this... what if we built on the plugin system idea we used for refgenie? What if divvy adapters could be provided via plugins? It could work basically the same way, but third parties, when installed, could populate some mapping that divvy could use.

Pro: this may actually eliminate the need to add adapters into the divvy file... the adapters would instead be provided by plugins
Con: the plugins would only be available within python. right now, theoretically, divvy templates could be used in another language. plugins could eliminate that.

To rescue this con, though... the plugin would only be useful for the tool that provided it anyway. So, maybe that's fine.

Con: The plugin actually requires information from the templates as well. because it needs to know what it's mapping TO, which is in the template. So, in a sense, the templates themselves are part of this, so it doesn't make sense for the plugin to be separate from divvy.

In other words, I think what I'm realizing, is that the divvy templates themselves really should be a separate thing that lives independent of both looper and divvy.
